src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -237,7 +237,9 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // not function properly.
             Href = output.Attributes[HrefAttributeName]?.Value as string;
 
-            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails);
+            var determinePartiallyMatchedAttributes = Logger.IsEnabled(
+                ModeMatchResultLoggerExtensions.PartiallyMatchedAttributesLogLevel);
+            var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails, determinePartiallyMatchedAttributes);
             Logger.TagHelperModeMatchResult(modeResult, context.UniqueId, ViewContext.View.Path, this);
```

Comment:
So assuming that you're using *our complicated attribute thing* and you *happen to look at logging* then this will give you a hint? Why is this better than a runtime exception if we know that what you're doing is wrong?

Think about what the experience would be like if we had time to build the tooling features we wanted here - would this be a compile error? warning? green squiggle? I don't think logging is a great surrogate for the experience that we really want.

I just want to make sure it's a good investment before we invest a lot of time into fixing the perf of this code.